### DISCUSSION TITLE
RS flipflop - Logic seems to oscillate

### DISCUSSION THREAD

**Discussion by kaviyarasu34** [hneemann/Digital#1112] [General]
Good day,
I am new to Logic seems to oscillate issue. How to solve this. It will be helpful for big circuits also. Attached the circuit image.
![rs-flip-flop](https://user-images.githubusercontent.com/41732151/217436872-11b45698-f81d-44e0-9120-2b53609dac42.png) . Thank you.

**Comment by mengstr** [comment]
What you have in the middle of the circuit is a NOR-based (an OR followed by a NOT is a NOR) S/R Latch and they oscillate if both inputs to it is zero. They do this in real life and in all simulators. So you must make sure that it never will happen.

Either you change the logic before it to prevent it from happening, or change the NOR S/R latch into a NAND S/R latch. It operates the same but oscillates when both inputs are high, and since the circuit you have now before the latch never can output two highs that would fix the problem.

**Reply by kaviyarasu34** [reply]
Thank you very much. I will try this method .
